src/anemoi/inference/grib/templates/file.py

```diff
@@ -31,23 +46,40 @@ def __init__(self, manager: Any, path: str) -> None:
             The manager instance.
         path : str
             The path to the GRIB file.
+        mode : Literal["auto", "first", "last"], optional
+            The method with which to select a message from the grib file to use as template, by default "first":
+            - "first": use the first message in the grib file
+            - "last": use the last message in the grib file
+            - "auto": select variable from the grib file matching the output variable name
+        variables : str | list, optional
+            The output variable name(s) for which to use this template file. If empty, applies to all variables.
         """
         self.manager = manager
-        self.path = path
+        self.path = Path(path)
+        self.mode = mode
+        self.variables = variables if isinstance(variables, list) else [variables] if variables else None
 
-    def template(self, grib: str, lookup: dict[str, Any]) -> ekd.Field:
-        """Retrieve the template from the GRIB file.
+    def __repr__(self):
+        info = f"{self.__class__.__name__}({self.path.name},mode={self.mode}{{variables}})"
+        return info.format(variables=f",variables={self.variables}" if self.variables else "")
 
-        Parameters
-        ----------
-        grib : str
```

Comment:
What do we gain by making this a cached property as opposed to just assigning an attribute inside the `__init__` (we would also know earlier if e.g. something goes wrong reading the data)?

EDIT: it's pointing to the wrong lines, the comment refers to the `_data` property.